core/arch/riscv/riscv.mk

```diff
@@ -0,0 +1,177 @@
+# Setup compiler for the core module
+ifeq ($(CFG_RV64_core),y)
+arch-bits-core := 64
+else
+arch-bits-core := 32
+endif
+CROSS_COMPILE_core := $(CROSS_COMPILE$(arch-bits-core))
+COMPILER_core := $(COMPILER)
+
+include mk/$(COMPILER_core).mk
+
+# Defines the cc-option macro using the compiler set for the core module
+include mk/cc-option.mk
+
+CFG_MMAP_REGIONS ?= 13
+CFG_RESERVED_VASPACE_SIZE ?= (1024 * 1024 * 10)
+
+ifeq ($(CFG_RV64_core),y)
+CFG_KERN_LINKER_FORMAT ?= elf64-littleriscv
+CFG_KERN_LINKER_ARCH ?= riscv
+else
+ifeq ($(CFG_RV32_core),y)
+CFG_KERN_LINKER_FORMAT ?= elf32-littleriscv
+CFG_KERN_LINKER_ARCH ?= riscv
+else
+$(error Error: CFG_RV64_core or CFG_RV32_core should be defined)
+endif
+endif
+
+CFG_CORE_RWDATA_NOEXEC ?= y
+CFG_CORE_RODATA_NOEXEC ?= n
+ifeq ($(CFG_CORE_RODATA_NOEXEC),y)
+$(call force,CFG_CORE_RWDATA_NOEXEC,y)
+endif
+
+core-platform-cppflags	+= -I$(arch-dir)/include
+core-platform-subdirs += \
+	$(addprefix $(arch-dir)/, kernel) $(platform-dir)
+
+# more convenient to move it to platform instead
+rv64-platform-cppflags += -mcmodel=medany -march=rv64imafd -mabi=lp64d
+rv64-platform-cppflags += -Wno-missing-include-dirs
+
+rv64-platform-cppflags += -DRV64=1 -D__LP64__=1
+rv32-platform-cppflags += -DRV32=1 -D__ILP32__=1
+
+platform-cflags-generic ?= -ffunction-sections -fdata-sections -pipe
+platform-aflags-generic ?= -pipe
+
+rv64-platform-cflags-generic := -mstrict-align $(call cc-option,-mno-outline-atomics,)
```

Comment:
`-mno-outline-atomics` makes no sense on RISC-V (it is related to the ARMv8.1-A instruction set).